### PR TITLE
refactoring unit-testing

### PR DESCRIPTION
**Pull request by mgherceanu**

### CODE REVIEW

File: src/pipes/by-vehicle-type/by-vehicle-type.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "deficiencyText": "insecure.",
+    "ref": "1.1.b",
+    "stdForProhibition": false,
+    "deficiencySubId": null,
+    "forVehicleType": ["psv", "hgv", "other"],
+    "deficiencyId": "b"
+  }];
+
+  it('return only psv items', () => {
+    expect((pipe.transform(defects, 'psv')).length).toBe(2);
```
Comment (danieliordache) [inline src/pipes/by-vehicle-type/by-vehicle-type.spec.ts]: use an enum for this

File: src/pipes/by-vehicle-type/by-vehicle-type.spec.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  it('return only psv items', () => {
+    expect((pipe.transform(defects, 'other')).length).toBe(1);
```
Comment (danieliordache) [inline src/pipes/by-vehicle-type/by-vehicle-type.spec.ts]: "other" is a valid option ?

File: src/pipes/by-vehicle-type/by-vehicle-type.spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { ByVehicleTypePipe } from "./by-vehicle-type";
+
+describe('ByVehicleTypePipe', () => {
+  let pipe = new ByVehicleTypePipe();
+  let defects = [{
```
Comment (danieliordache) [inline src/pipes/by-vehicle-type/by-vehicle-type.spec.ts]: use the mock data

File: src/providers/defects/defects.service.spec.ts [hunk 4]
```diff
@@ -57,7 +57,25 @@ describe('Provider: DefectsService', () => {
   });
 
   it('should return badge\'s correct color', () => {
-    expect(defectsService.getBadgeColor('major')).toBe('danger')
+    expect(defectsService.getBadgeColor('minor')).toBe('attention');
```
Comment (danieliordache) [inline src/providers/defects/defects.service.spec.ts]: use an enum

File: src/providers/defects/defects.service.spec.ts [hunk 5]
```diff
@@ -57,7 +57,25 @@ describe('Provider: DefectsService', () => {
...
+    expect(defectsService.getBadgeColor('major')).toBe('danger');
+    expect(defectsService.getBadgeColor('dangerous')).toBe('dark');
+    expect(defectsService.getBadgeColor('advisory')).toBe('light');
+    expect(defectsService.getBadgeColor('prs')).toBe('tertiary');
+  });
+
+  it('should return list if nothing entered', () => {
+    let filter = null;
+    let filteredData = defectsService.searchDefect(defectsCategories, filter, ['imNumber', 'imDescription']);
```
Comment (danieliordache) [inline src/providers/defects/defects.service.spec.ts]: create an enum for those

File: src/providers/defects/defects.service.spec.ts [hunk 6]
```diff
@@ -107,4 +125,55 @@ describe('Provider: DefectsService', () => {
...
+    expect(defect.additionalInformation.location).toBeDefined();
+    expect(defect.stdForProhibition).toBeFalsy();
+    expect(defect.metadata).toBeTruthy();
+    expect(defect.prs).toBeFalsy();
+  });
+
+  it('should sort an aray based on key and order', () => {
+    let defectsArray = defectsCategories;
+    expect(defectsArray[0].imDescription).toContain('Registration Plate');
+    let sortedDefectsArray = defectsService.orderDefectsArray(defectsArray, 'imDescription', 'desc');
```
Comment (danieliordache) [inline src/providers/defects/defects.service.spec.ts]: create an enum for those

File: src/providers/global/auth.service.spec.ts [hunk 7]
```diff
@@ -40,5 +46,50 @@ describe(`AuthService`, () => {
...
+  });
+
+  it('should set the tester details', () => {
+    let details;
+    expect(details).toBeUndefined();
+    details = authService.setTesterDetails(null);
+    expect(details).toBeTruthy();
+  });
+
+  it('should set the tester details', () => {
```
Comment (danieliordache) [inline src/providers/global/auth.service.spec.ts]: use the token from the auth mock

File: src/providers/global/http.service.spec.ts [hunk 8]
```diff
@@ -44,6 +45,85 @@ describe(`Provider: HttpService`, () => {
...
+  it('test postTestResult', () => {
+    let data;
+    expect(data).toBeUndefined();
+    data = httpService.postTestResult('23423443');
+    expect(data).toBeTruthy();
+  });
+
+  it('test startVisit', () => {
+    let data;
+    let activities: ActivityModel = {
```
Comment (danieliordache) [inline src/providers/global/http.service.spec.ts]: use/create a mock for it

File: src/providers/test-type/test-type.service.spec.ts [hunk 9]
```diff
@@ -47,6 +50,86 @@ describe('Provider: TestTypeService', () => {
...
+    expect(result).toMatch(TEST_TYPE_RESULTS.ABANDONED);
+  });
+
+  it('should ordered testType array', () => {
+    let testTypes: TestTypeModel[] = [];
+    testTypes.push(TestTypeDataModelMock.TestTypeData);
+    testTypes.push(TestTypeDataModelMock.TestTypeData);
+    testTypes[0].testTypeId = '2';
+    expect(testTypes[0].testTypeId).toMatch('2');
+    let orderedTestTypes: TestTypeModel[] = testTypeService.orderTestTypesArray(testTypes, 'testTypeId', 'asc');
```
Comment (danieliordache) [inline src/providers/test-type/test-type.service.spec.ts]: create an enum for those

File: src/providers/utils/common-functions.spec.ts [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+  it('should return elements from array', () => {
+    let searchVal = 'bg';
+    let array = CountryOfRegistrationData.CountryData;
+    let filteredArray = commonFunctionsService.searchFor(array, searchVal, ['key', 'value']);
+    expect(filteredArray[0]['key']).toMatch('bg');
+  });
+
+  it('should return empty array if not found', () => {
+    let searchVal = '123';
+    let array = [
```
Comment (danieliordache) [inline src/providers/utils/common-functions.spec.ts]: use/create a mock for those

File: src/providers/utils/common-functions.spec.ts [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+
+  it('should return an array ordered by stringId, order given', () => {
+    let array = TestTypesReferenceDataMock.TestTypesData;
+    let sortedArray = array.sort(commonFunctionsService.orderByStringId('id', 'desc'));
+    expect(sortedArray[0]['id']).toMatch('5');
+  });
+
+  it('should make a clone of an given object', () => {
+    let object = {
+      name: 'Grasu XXL',
```
Comment (danieliordache) [inline src/providers/utils/common-functions.spec.ts]: change this

File: src/providers/visit/visit.service.spec.ts [hunk 12]
```diff
@@ -54,30 +54,49 @@ describe('Provider: VisitService', () => {
...
-    expect(storageService.delete).toHaveBeenCalled();
+    visitService.endVisit('f34f43');
+    expect(httpService.endVisit).toHaveBeenCalled();
+  });
+
+  it('should return the latest test', () => {
+    visitService.createVisit(TEST_STATION);
+    let aTest: TestModel = {
+      "startTime": null,
+      "endTime": '14 Martie',
```
Comment (danieliordache) [inline src/providers/visit/visit.service.spec.ts]: change to english

File: src/pipes/by-vehicle-type/by-vehicle-type.spec.ts (hunk 1 above)
Comment (mgherceanu) [reply src/pipes/by-vehicle-type/by-vehicle-type.spec.ts]: done

File: src/pipes/by-vehicle-type/by-vehicle-type.spec.ts (hunk 2 above)
Comment (mgherceanu) [reply src/pipes/by-vehicle-type/by-vehicle-type.spec.ts]: no, it is just an option added to the test in order to filter by it and receive 1 element, all the test had psv and hgv

File: src/pipes/by-vehicle-type/by-vehicle-type.spec.ts (hunk 3 above)
Comment (mgherceanu) [reply src/pipes/by-vehicle-type/by-vehicle-type.spec.ts]: done

File: src/providers/defects/defects.service.spec.ts (hunk 4 above)
Comment (mgherceanu) [reply src/providers/defects/defects.service.spec.ts]: done

File: src/providers/defects/defects.service.spec.ts (hunk 6 above)
Comment (mgherceanu) [reply src/providers/defects/defects.service.spec.ts]: for categories or some strings from the reference data? the only usage is her in this test,

File: src/providers/defects/defects.service.spec.ts (hunk 5 above)
Comment (mgherceanu) [reply src/providers/defects/defects.service.spec.ts]: for categories or some strings from the reference data? the only usage is her in this test

File: src/providers/global/auth.service.spec.ts (hunk 7 above)
Comment (mgherceanu) [reply src/providers/global/auth.service.spec.ts]: i use some mocked auth response to testauthService.setTesterDetails(authResponse);
i am using the token u have mocked at the beginning

File: src/providers/global/http.service.spec.ts (hunk 8 above)
Comment (mgherceanu) [reply src/providers/global/http.service.spec.ts]: done

File: src/providers/test-type/test-type.service.spec.ts (hunk 9 above)
Comment (mgherceanu) [reply src/providers/test-type/test-type.service.spec.ts]: for what exactly ? the testType property name? and for 'asc' ?

File: src/providers/utils/common-functions.spec.ts (hunk 11 above)
Comment (mgherceanu) [reply src/providers/utils/common-functions.spec.ts]: done

File: src/providers/utils/common-functions.spec.ts (hunk 10 above)
Comment (mgherceanu) [reply src/providers/utils/common-functions.spec.ts]: i just needed a small array and different of the countries array in order to cover all test cases

File: src/providers/visit/visit.service.spec.ts (hunk 12 above)
Comment (mgherceanu) [reply src/providers/visit/visit.service.spec.ts]: done

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts [hunk 13]
```diff
@@ -36,6 +36,7 @@ export class TestTypesListPage implements OnInit {
 
   ngOnInit() {
     if(this.testTypeReferenceData) this.testTypeReferenceData = this.testTypeService.orderTestTypesArray(this.testTypeReferenceData, 'id', 'asc')
+    console.log(JSON.stringify(this.testTypeReferenceData));
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-types-list/test-types-list.ts]: please remove this console.log

Comment (mgherceanu) [reply src/pages/testing/test-creation/test-types-list/test-types-list.ts]: done

File: sonar-project.properties [hunk 14]
```diff
@@ -1,5 +1,5 @@
 #----- Default SonarQube server
-sonar.host.url=http://volted-monkey-sonarqube.ops:9000
+sonar.host.url=http://localhost:9000
```
Comment (vladddragomir) [inline sonar-project.properties]: please change this as it was before (volted-monkey...)

Comment (vladddragomir) [reply sonar-project.properties]: updated

**Pull request merged**